geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java

```diff
@@ -680,37 +695,85 @@ private void closeListener(TransactionListener tl) {
    *         to reactivate the suspended transaction.
    */
   public TXStateProxy internalSuspend() {
+    return internalSuspend(false);
+  }
+
+  /**
+   * If the current thread is in a transaction then suspend will cause it to no longer be in a
+   * transaction.
+   *
+   * @param needToResumeBySameThread whether a suspended transaction needs to be resumed by the same
+   *        thread.
+   * @return the state of the transaction or null. Pass this value to {@link TXManagerImpl#resume}
```

Comment:
This comment is wrong. I think you need to pass this returned value to internalResume not resume